### ISSUE TITLE
Modal is slow when trigger event is added to many dynamically generated buttons

### ISSUE THREAD

**Issue by kunokdev** [Semantic-Org/Semantic-UI#4852]
Hello, I have an performance issue with Semantic UI modal component.
I have ~300 dynamic elements on my page (card element) and each has a button.
With my code below I added event listener to each button that would trigger same modal.

```
var buttons = document.querySelectorAll('.button[data-for-modal="detailed-info"]');
     buttons.forEach(function(button){
        button.addEventListener('click', function(){
            $('.modal[data-modal="detailed-info"]').modal('setting','transition','vertical flip').modal('show');
          })
      })
```

It is really slow. Takes around 1 sec on desktop and more than 3 seconds on mobile to display modal.
If I would replace modal method with for example alert('test'), it would not lack performance. It would show alert within millisecond. So issue must be related to modal.

Why is it so slow and how do I make it faster?

**Comment by IonutBajescu** [comment]
Hey,

Can you please explain what exactly takes three seconds? If it's adding the event listeners, then please take a look at this article: https://www.kirupa.com/html5/handling_events_for_many_elements.htm

**Comment by kunokdev** [comment]
@IonutBajescu Hello, the modal part. So If I would replace the modal part with alert('test') it would call alert function instantly

**Comment by IonutBajescu** [comment]
I see, any chance you can push the code to jsfiddle.com? 1 to 3 seconds seems a little too much for opening a bare modal, although I could see the transitions acting incorrectly in some environments.

**Comment by kunokdev** [comment]
@IonutBajescu I have tried to recreate implementation of this in fiddle but it failed.[]( http://codepen.io/kunokdev/pen/VmOLvr). My page has more child nodes than in this example, not sure if it's related.

**Comment by kunokdev** [comment]
Ok I did a bit more testing. 

This is not directly related to Semantic UI anymore.

I've tried doing custom modal (so this time I am not using alert, but I am manipulating DOM) and it's slow again.

So the issue is huge DOM tree that slows things down.

I've tried this same code on example where there are less results (~100) and it's not slow anymore.

I am going to do my best to improve performance, and if that won't be enough, pagination will do.
